docs/Benchmark-project.md

```diff
@@ -0,0 +1,84 @@
+
+2. **Reproducible and self-service** - The benchmark code should allow users to reproduce the results on their end, or adapt it to another environment to run the full benchmark on their data.
+
+3. **Act as reference** - The scripts provided in the benchmark could be used as reference / standard implementations in python for training and inferencing. The code should be readable, documented and unit tested. It should make the best and up-to-date use of the python libraries it proposes to evaluate.
+
+## Running the benchmark
+
+We propose the benchmark to be available both for manual or orchestrated (cloud-based) run.
+
+- **Manual** - Users can reproduce the benchmark results while running the code locally to a given VM.
+
+- **Orchestrated** - Users can run the benchmark in a cloud-based ML orchestration platform (ex: AzureML).
+
+We also want to identify how to enable users to run the benchmark on custom lightgbm builds (either local or from a github branch), so that they can assess the performance of their changes, or use different compilation options.
+
+## Data and tasks
+
+The users should be able to run the benchmark on multiple data sources. Not only we want to provide synthetic and public data benchmarks, but we also propose to make sure users can use the benchmark on their on data sets. This can be facilitated by using a cloud-based ML platform like AzureML.
+
+- **Synthetic Data** - Generated fake data, in particular to test scalability, inferencing latency.
+
+- **Public Data** - Example public datasets used in the community and literature to benchmark the learners.
+
+- **Production Data** - Provide the ability for users to run the benchmark on their own data.
+
+## Code Structure
+
+To allow for modularity and standardization, we propose to structure the benchmark into a collection of independent python scripts (specs in a separate document). Each script will have its own environment, inputs, outputs and parameters. All scripts will eventually draw from common helper classes to support implementation routines (reporting, argument parsing, etc).
+
+## Phases and Milestones
+
+We propose the following process to iterate on implementation.
+
+### 1. Foundations phase
+
+In this first phase, we'll implement the foundations to allow the benchmark to grow later. We'll start with:
+- a couple minimal scripts for generating synthetic data, training lightgbm and inferencing
+- some common libraries around metrics reporting
+- a first benchmark pipeline in either AzureML SDK1.5 or SDK2.0 to run end-to-end and report metrics
+- enabling both manual and orchestrated runs of the benchmark
+- a build with a first set of unit tests
+- documentation of the goals, developer guide and common routines
+
+**Milestone**: an end-to-end run of the benchmark both locally (VM) and in the cloud, reporting a minimal set of wall time metrics, running on synthetic data, producing numbers we manually report in the repo as markdown.
```

Comment:
Suggest writing explicitly here or somewhere else &mdash; what information will go with those metrics.

Hopefully:

- "pointer" to data
- VM SKU
- ???

Reply:
Yep, I will elaborate on those in the reporting PR.